mountpoint-s3-fs/benches/cache_serialization.rs

```diff
@@ -0,0 +1,68 @@
+use std::fs;
+
+use criterion::async_executor::{AsyncExecutor, FuturesExecutor};
+use criterion::{criterion_group, criterion_main, Criterion};
+
+use mountpoint_s3_client::types::ETag;
+use mountpoint_s3_fs::data_cache::{ChecksummedBytes, DataCache, DiskDataCache, DiskDataCacheConfig};
+use mountpoint_s3_fs::object::ObjectId;
+use rand::RngCore;
+use tempfile::TempDir;
+
+const BLOCK_SIZE: u64 = 1024 * 1024;
+const OBJECT_SIZE: usize = 10 * BLOCK_SIZE as usize;
+
+#[inline]
+async fn read_cache_block(cache: &DiskDataCache, cache_key: &ObjectId) {
+    _ = cache
```

Comment:
consider using `std::hint::black_box` on the result to ensure the read is not optimised out